### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by devsnd**
Changed Oracle and Oracle signature so that you have to use their constructor instead of using class attributes to initialize them.

General code cleanup including tests.

### CODE REVIEW

File: aeternity/epoch.py [hunk 1]
```diff
@@ -180,7 +180,7 @@ def run(self):
             return
 
 
-class EpochComponent(ValidateClassMixin):
+class EpochSubscription():
```
Comment (tillkolter) [inline aeternity/epoch.py]: Good decision! I felt that something was not ideal with the last naming, but wasn't aware of that until now :)

File: aeternity/oracle.py [hunk 2]
```diff
@@ -173,23 +197,14 @@ def handle_registration_response(self, message):
...
-            'action': 'register',
-            'origin': 'oracle',
-            'payload': {
-                'result': 'ok',
-                'oracle_id': 'ok$3WRqCYwdr9B5aeAMT7Bfv2gGZpLUdD4RQM4hzFRpRzRRZx7d8pohQ6xviXxDTLHVwWKDbGzxH1xRp19LtwBypFpCVBDjEQ',
-                'tx_hash': 'th$bRFKbTNEMhUJE23G6d7XLkrKcivaDuH6nihkKmqQPzxp4nocz'
-            }
-        }
-        oracle_id = message['payload'].get('oracle_id')
+    def handle_oracle_subscription_ack(self, message):
```
Comment (tillkolter) [inline aeternity/oracle.py]: is `ack` for acknowledgement widely accepted knowledge that it can be intuitively understood?

**Comment by tillkolter** [comment]
There was not much space here for improving comments. 

Only thing that I would recommend to give a second thought is the naming for `handle_oracle_subscription_ack `. I cant really put into the right terms or recommend anything better, it just sounds a little cumbersome.

File: aeternity/oracle.py (hunk 2 above)
Comment (devsnd) [reply aeternity/oracle.py]: yeah, it's really common in comp-sci lingo, so I'd just roll with it.

**Pull request merged**